apiserver/failoverservice/failoverimpl.go

```diff
@@ -243,13 +190,16 @@ func validateDeploymentName(deployName, clusterName, ns string) (*v1.Deployment,
 	}
 
 	//make sure the primary is not being selected by the user
```

Comment:
This is more of a comment about the function over all and not necessarily the changes at hand. The function name is a bit confusing, IMO. Could we add some documentation to clear up what this function is doing?